pages/index.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <ModalAddNewEvent :date="date" @closeModal="closeModal" v-if="modal" />
+  <div class="root-container">
+    <Calender
+      @openModal="handleModal"
+      @toggleModal="toggleModal"
+      :Allevents="userCreatedEvent"
+      :key="update_component"
+      :selectedEvents="selectedEvents"
+    />
+    <!-- calenderRightbar -->
+    <EventNavBar :events="events" @include="addEvent" @exclude="removeEvent" />
+  </div>
+</template>
+
+<script setup>
+const modal = ref(false);
+const date = ref(null);
+const userCreatedEvent = ref([]);
+const update_component = ref(false);
+const events = ref([
+  {
+    name: "Holiday",
+    image: "/images/Holiday.png",
+    id: 1,
+  },
+  {
+    name: "Birthday",
+    image: "/images/BirthDay.png",
+    id: 2,
+  },
+  {
+    name: "Leave",
+    image: "/images/leave.png",
+    id: 3,
+  },
+]);
+const selectedEvents = ref(events.value.map((item) => item.id));
+
+const handleModal = (dateOfClick) => {
+  date.value = dateOfClick;
+  modal.value = true;
+};
+const color = ["#F87171", "#FBBF24", "#34D399", "#60A5FA"];
+
+const closeModal = (prop = null) => {
+  if (prop === null) {
+    modal.value = false;
+    return;
+  }
+
+  userCreatedEvent.value.push({
+    what: prop.what,
+    eventUser: prop.eventUser,
+    id: prop.eventType,
+    backgroundColor: color[prop.eventType],
+    startDate: prop.startDate,
+    endDate: prop.endDate,
+  });
+  console.log(userCreatedEvent.value);
+  update_component.value = !update_component.value;
+  modal.value = false;
+};
+
+const addEvent = (prop) => {
```

Comment:
don't name the argument `prop`. rename it to `eventID` as we are sending event.id from `EventNavBar.vue`